File: RxRetrofitApp/japp/src/main/java/com/woowrale/rxretrofit/japp/data/repository/ApiServices.java
package com.woowrale.rxretrofit.japp.data.repository;

import com.woowrale.rxretrofit.japp.data.model.Crypto;
import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface ApiServices {

    @GET("{coin}-usd")
    Observable<Crypto> getCoinData(@Path("coin") String coin);
}

File: AndroidXPatternsApp/kmvvm-app/src/main/java/com/woowrale/kmvvm/data/repository/ApiServices.kt
package com.woowrale.kmvvm.data.repository

import com.woowrale.kmvvm.data.model.Contact
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.Url

interface ApiContacts {

    @GET
    fun getContacts(
        @Url apiContacts: String,
        @Query("source") source: String,
        @Query("search") query: String
    ): Single<List<Contact>>

}